tests/test_runner/test_discover_runner.py

```diff
@@ -122,6 +122,13 @@ def test_name_patterns(self):
                 ).build_suite(['test_runner_apps.simple'])
                 self.assertEqual(expected, self.get_test_methods_names(suite))
 
+    def test_loader_pattern_not_mutated(self):
+        runner = DiscoverRunner(test_name_patterns=['*_tests.py'], verbosity=0)
```

Comment:
The `test_name_patterns` argument shouldn't normally end in `.py` (that's `pattern`). So you can just pass `['UnittestCase1']` via this argument.